cloudinit/sources/DataSourceAzure.py

```diff
@@ -554,6 +554,35 @@ def _is_ephemeral_networking_up(self) -> bool:
             or self._ephemeral_dhcp_ctx.lease is None
         )
 
+    def _check_azure_proxy_agent_status(self) -> None:
+        """check if azure-proxy-agent is fully running and can procede to
```

Comment:
nit: would encourage following https://peps.python.org/pep-0257/ and https://peps.python.org/pep-0008/#documentation-strings

and do a short one-liner followed by a longer paragraph.

"""Check if azure-proxy-agent is ready for communication to Wireserver/IMDS.

If ProvisionGuestProxyAgent is true, query azure-proxy-agent status, waiting up to 120 seconds for the proxy to negotiate with Wireserver and configure an eBPF proxy.  Once azure-proxy-agent is ready, it will exit with code 0 and cloud-init can then expect to be able to communicate with these services.

Fail deployment if azure-proxy-agent is not found or otherwise returns an error.

For more information, check out:
https://github.com/azure/guestproxyagent
"""
```